### PR TITLE
Add process to collects some stats

### PR DESCRIPTION
**Pull request by skrakau**
.. collect number of unique proteins, total peptides and unique peptides per condition as well as the number of unique peptides across conditions.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/metapep branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/metapep)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ Initial release of nf-core/metapep, created with the [nf-core](https://nf-co.re/
 - [#60](https://github.com/skrakau/metapep/pull/60) - Add entities to data model
 - [#60](https://github.com/skrakau/metapep/pull/60) - Add option for `bins` input type
 - [#60](https://github.com/skrakau/metapep/pull/60) - Add plotting of entity-wise binder ratios
+- [#64](https://github.com/skrakau/metapep/pull/64) - Add process to collect some stats (`results/db_tables/stats.txt`)
```
Comment (lkuchenb) [inline CHANGELOG.md]:
```suggestion
- [#64](https://github.com/skrakau/metapep/pull/64) - Add process to collect stats (`results/db_tables/stats.txt`)
```

File: bin/collect_stats.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    # condition_name, peptide_id, condition_peptide_count
+    conditions_peptides = results\
+            .merge(protein_peptide_occs)\
+            .drop(columns="protein_id")\
+            .groupby(["condition_name", "peptide_id"])["count"]\
+            .sum()\
+            .reset_index(name="condition_peptide_count")
+
+    # condition_name, unique_peptide_count
```
Comment (lkuchenb) [inline bin/collect_stats.py]:
```suggestion
    # condition_name, total_peptide_count
```

File: bin/collect_stats.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    # Read input files
+    peptides                  = pd.read_csv(args.peptides, sep='\t')
+    protein_peptide_occs      = pd.read_csv(args.protein_peptide_occ, sep='\t')
+    entities_proteins_occs    = pd.read_csv(args.entities_proteins_occ, sep='\t')
+    microbiomes_entities_occs = pd.read_csv(args.microbiomes_entities_occ, sep='\t').drop(columns="entity_weight")
+    conditions                = pd.read_csv(args.conditions, sep='\t').drop(columns="condition_id")
+
+    print("Joining input data...", flush=True)
+
+    results = conditions\
```
Comment (lkuchenb) [inline bin/collect_stats.py]: `results` confused me. Maybe `conditions_proteins`?

File: conf/base.config [hunk 4]
```diff
@@ -36,6 +36,12 @@ process {
     time = { check_max( 20.h * task.attempt, 'time' ) }
     cache = 'lenient'
   }
+  withName:collect_stats {
+    cpus = { check_max( 1 * task.attempt, 'cpus' ) }
+    memory = { check_max( 200.GB * task.attempt, 'memory' ) }
```
Comment (lkuchenb) [inline conf/base.config:41]: Does this process really need that much mem? It doesn't read the predictions at all.

**Review by lkuchenb — APPROVED** [review]

File: conf/base.config (hunk 4 above)
Comment (skrakau) [reply conf/base.config:41]: Probably not, but I would rather adjust/decrease the parameters afterwards.

**Comment by skrakau** [comment]
Thanks!

**Pull request merged**